_search-plugins/concurrent-segment-search.md

```diff
@@ -0,0 +1,168 @@
+
+### Enable in opensearch.yml
+
+If you are running an OpenSearch cluster and want to enable concurrent segment search in the config file, add the following line to `opensearch.yml`:
+
+```yaml
+opensearch.experimental.feature.concurrent_segment_search.enabled=true
+```
+{% include copy.html %}
+
+### Enable with Docker containers
+
+If you’re running Docker, add the following line to `docker-compose.yml` under the `opensearch-node` > `environment` section:
+
+```bash
+OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.concurrent_segment_search.enabled=true"
+```
+{% include copy.html %}
+
+### Enable on a node using a tarball install
+
+To enable concurrent segment search on a tarball install, provide the new JVM parameter either in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS`.
+
+#### OPTION 1: Modify jvm.options
+
+Add the following lines to `config/jvm.options` before starting the OpenSearch process to enable the feature and its dependency:
+
+```bash
+-Dopensearch.experimental.feature.concurrent_segment_search.enabled=true
+```
+{% include copy.html %}
+
+Then run OpenSearch:
+
+```bash
+./bin/opensearch
+```
+{% include copy.html %}
+
+#### OPTION 2: Enable with an environment variable
+
+As an alternative to directly modifying `config/jvm.options`, you can define the properties by using an environment variable. This can be done in a single command when you start OpenSearch or by defining the variable with `export`.
+
+To add these flags in-line when starting OpenSearch, run the following command:
+
+
+```bash
+OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.concurrent_segment_search.enabled=true" ./opensearch-{{site.opensearch_version}}/bin/opensearch
+```
+{% include copy.html %}
+
+If you want to define the environment variable separately, prior to running OpenSearch, run the following commands:
+
+```bash
+export OPENSEARCH_JAVA_OPTS="-Dopensearch.experimental.feature.concurrent_segment_search.enabled=true"
+```
+{% include copy.html %}
+
+```bash
+./bin/opensearch
+```
+{% include copy.html %}
+
+## Disabling concurrent search for an index or all indexes
+
+After you enable the experimental feature flag, all search requests will use concurrent segment search during query phase. To disable concurrent segment search for all indexes, set following dynamic cluster setting:
+
+
+```json
+PUT _cluster/settings
+{
+   "persistent":{
+      "search.concurrent_segment_search.enabled": false
+   }
+}
+```
+{% include copy-curl.html %}
+
+To disable concurrent segment search for a particular index, specify the index name in the endpoint:
+
+```json
+PUT <index-name>/_settings
+{
+    "index.search.concurrent_segment_search.enabled": false
+}
+```
+{% include copy-curl.html %}
+
+## Concurrent segment search mechanisms
+
+You can choose one of the two available concurrent segment mechanisms: the default [Lucene mechanism](#the-lucene-mechanism) or the [max slice count mechanism](#the-max-slice-count-mechanism).
```

Comment:
These mechanisms are about controlling the amount of concurrency, so maybe the section should be titled "Mechanisms to control search concurrency" or something along those lines.